libcudacxx/include/cuda/std/__cccl/diagnostic.h

```diff
@@ -92,9 +92,10 @@
     _CCCL_DIAG_SUPPRESS_GCC("-Wdeprecated-declarations")
 #  define _CCCL_SUPPRESS_DEPRECATED_POP _CCCL_DIAG_POP
 #elif _CCCL_COMPILER(NVHPC)
-#  define _CCCL_SUPPRESS_DEPRECATED_PUSH \
-    _CCCL_DIAG_PUSH                      \
-    _CCCL_DIAG_SUPPRESS_NVHPC(deprecated_entity)
+#  define _CCCL_SUPPRESS_DEPRECATED_PUSH         \
+    _CCCL_DIAG_PUSH                              \
+    _CCCL_DIAG_SUPPRESS_NVHPC(deprecated_entity) \
+    _CCCL_DIAG_SUPPRESS_NVHPC(deprecated_entity_with_custom_message)
```

Comment:
I'm planning to carve these out into a separate PR. But, for the time being, using this PR to explore all the remaining issues that some compilers have with suppressing deprecation warnings.